library/core/src/ptr/const_ptr.rs

```diff
@@ -1768,6 +1768,46 @@ impl<T> *const [T] {
     }
 }
 
+impl<T, const N: usize> *const [T; N] {
+    /// Returns a raw pointer to the array's buffer.
+    ///
+    /// This is equivalent to casting `self` to `*const T`, but more type-safe.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// #![feature(array_ptr_get)]
+    /// use std::ptr;
+    ///
+    /// let arr: *const [i8; 3] = ptr::null();
+    /// assert_eq!(arr.as_ptr(), ptr::null());
+    /// ```
+    #[inline]
+    #[unstable(feature = "array_ptr_get", issue = "119834")]
+    #[rustc_const_unstable(feature = "array_ptr_get", issue = "119834")]
+    pub const fn as_ptr(self) -> *const T {
+        self as *const T
+    }
+
+    /// Returns a raw pointer to a shared slice containing the entire array.
```

Comment:
@Nilstrieb should this be "a non-mutable" slice, as in your other suggestion? Or "const"?

Reply:
great question, i'm not sure either. "shared slice" feels a bit wrong, but "const" is not exactly great either. also not a fan of "non-mutable"  :sweat_smile: 
Maybe just drop the qualifier and just say "to a slice"?